### PR TITLE
constraints, create table: improve clarity and update out of date info

### PR DESCRIPTION
**Pull request by nullnotnil**
### What is changed, added or deleted? (Required)

The statement reference for `CREATE TABLE` is a little bit weird:
* As well as having a `synopsis` sub heading, it also had a `syntax` sub heading.
* Many of the items under the `syntax` subheading describe options that are not actually supported, but parsed and ignored. As such, it feels weird to document these.
* Some non-critical info such as "doesn't include check constraints" and "supports some foreign key constraints" are also under `syntax` - these are best suited for the "MySQL compatibility" subheading. The point about foreign keys was also incorrect - it is not supported in syntax only, since this meta data was always stored, and relationships were validated. It is not yet enforced by DML is all.
* LIKE is documented, but it also explicitly says that there is a separate docs page for like.
* It incorrectly said utf8mb4 is the only supported character set.

In addition to this, I have added `CHECK` constraints to the constraints page, and removed a reference to FOREIGN KEYS not displayed in show create table (version specific change, this is fixed in master only: https://github.com/pingcap/tidb/pull/18360 ).

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [x] Might cause conflicts

### CODE REVIEW

File: sql-statements/sql-statement-create-table.md [hunk 1]
```diff
@@ -70,170 +70,16 @@ This statement creates a new table in the currently selected database. See also
...
 | Parameters | Description | Example |
 | ---------- | ---------- | ------- |
 | `AUTO_INCREMENT` | The initial value of the increment field | `AUTO_INCREMENT` = 5 |
-|`SHARD_ROW_ID_BITS`| To set the number of bits for the implicit `_tidb_rowid` shards |`SHARD_ROW_ID_BITS` = 4|
+| [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)| To set the number of bits for the implicit `_tidb_rowid` shards |`SHARD_ROW_ID_BITS` = 4|
 |`PRE_SPLIT_REGIONS`| To pre-split `2^(PRE_SPLIT_REGIONS)` Regions when creating a table |`PRE_SPLIT_REGIONS` = 4|
 |`AUTO_ID_CACHE`| To set the auto ID cache size in a TiDB instance. By default, TiDB automatically changes this size according to allocation speed of auto ID |`AUTO_ID_CACHE` = 200|
 |`AUTO_RANDOM_BASE`| To set the initial incremental part value of auto_random. This option can be considered as a part of the internal interface. Users can ignore this parameter |`AUTO_RANDOM_BASE` = 0|
-| `CHARACTER SET` | To specify the string code for the table; currently only support UTF8MB4 | `CHARACTER SET` =  'utf8mb4' |
+| `CHARACTER SET` | To specify the character set for the table | `CHARACTER SET` =  'utf8mb4' |
```
Comment (kolbe) [inline sql-statements/sql-statement-create-table.md:82]: Maybe "character set" could be a link to https://docs.pingcap.com/tidb/dev/character-set-and-collation?

File: constraints.md [hunk 2]
```diff
@@ -58,6 +58,25 @@ Query OK, 1 row affected (0.03 sec)
 * The second `INSERT` statement fails because the `age` column is defined as `NOT NULL`.
 * The third `INSERT` statement succeeds because the `last_login` column is not explicitly defined as `NOT NULL`. NULL values ​​are allowed by default.
 
+## CHECK
+
+TiDB parses but ignores `CHECK` constraints. This is MySQL 5.7 compatible behavior.
+
+For example:
+{{< copyable "sql" >}}
```
Comment (lilin90) [inline constraints.md:66]:
```suggestion

{{< copyable "sql" >}}
```

File: sql-statements/sql-statement-create-table.md [hunk 3]
```diff
@@ -269,20 +135,55 @@ mysql> SELECT * FROM t1;
...
+```sql
+DROP TABLE IF EXISTS t1;
+CREATE TABLE IF NOT EXISTS t1 (
+ id BIGINT NOT NULL PRIMARY KEY auto_increment,
+ b VARCHAR(200) NOT NULL
+);
+DESC t1;
+```
+
+```
```
Comment (lilin90) [inline sql-statements/sql-statement-create-table.md:151]:
```suggestion
```sql
```

File: sql-statements/sql-statement-create-table.md [hunk 4]
```diff
@@ -70,170 +70,16 @@ This statement creates a new table in the currently selected database. See also
...
-  | MAX_ROWS [=] value
-  | MIN_ROWS [=] value
-  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
-  | STATS_PERSISTENT [=] {DEFAULT|0|1}
-```
-
-The `table_option` currently only supports `AUTO_INCREMENT`, [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md), `PRE_SPLIT_REGIONS`, `CHARACTER SET`, `COLLATE`, and `COMMENT`, while the others are only supported in syntax. The clauses are separated by a comma `,`. See the following table for details:
+The following *table_options* are supported. Other options such as `AVG_ROW_LENGTH`, `CHECKSUM`, `COMPRESSION`, `CONNECTION`, `DELAY_KEY_WRITE`, `ENGINE`, `KEY_BLOCK_SIZE`, `MAX_ROWS`, `MIN_ROWS`, `ROW_FORMAT` and `STATS_PERSISTENT` are parsed but ignored.
 
 | Parameters | Description | Example |
```
Comment (lilin90) [inline sql-statements/sql-statement-create-table.md:75]:
```suggestion
| Options | Description | Example |
```

**Pull request merged**